commands.py

```diff
@@ -20,9 +23,121 @@ def __init__(self, bot):
         Args:
             bot: The discord bot
         """
+
         self.bot = bot
         self.datasource = JsonDatasourceHandler()
 
+    @tasks.loop(minutes=5)
+    async def check_streamers(self):
+        """
+        Checks twitch streamers every interval to be able to announce go lives
+
+        Returns:
+            None
+        """
+
+        streamer_mapper = StreamerMapper(self.datasource)
+        streamers = streamer_mapper.map()
+        twitch_handler = TwitchHandler()
+        live_streams = twitch_handler.get_streams(streamers)
+
+        if len(live_streams) > 0:
+            streamers = await self.mark_as_online(streamers, live_streams)
+
+        self.mark_as_offline(streamers, live_streams)
+
+    async def announce(self, streamer, stream) -> None:
+        """
+        Announces to the discord server when a streamer goes live
+
+        Args:
+            streamer (StreamerInterface): The streamer
+            stream (TwitchStreamInterface): The twitch stream
+
+        Returns:
+            None
+        """
+
+        load_dotenv()
+        channel = self.bot.get_channel(os.getenv('DISCORD_ANNOUNCE_CHANNEL'))
+        user = await self.bot.fetch_user(streamer.id)
+        embed = discord.Embed(
+            title=streamer.username,
+            description=f"{user.mention} has just gone live on Twitch!"
+        )
+
+        if stream.thumbnail:
+            thumbnail = stream.thumbnail
+            thumbnail = thumbnail.replace('{width}', '600')
+            thumbnail = thumbnail.replace('{height}', '400')
+            embed.set_image(url=thumbnail)
+
+        embed.add_field(name="Twitch URL", value=f'https://twitch.tv/{streamer.username}')
+
+        if stream.title:
+            embed.add_field(name="Title", value=stream.title, inline=False)
+
+        if stream.game_name:
+            embed.add_field(name="Currently Playing", value=stream.game_name, inline=False)
+
+        if len(streamer.roles) > 0:
```

Comment:
We use this type of functionality in two places, maybe create a static method for it.